src/net.cpp

```diff
@@ -2672,14 +2680,17 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
                     // a currently-connected peer.
                     addrman.Good(addr);
                     // Select a new table address for our feeler instead.
-                    std::tie(addr, addr_last_try) = addrman.Select(true);
+                    std::tie(addr, addr_last_try) = addrman.Select(true, networks);
                 }
             } else {
                 // Not a feeler
                 // If preferred_net has a value set, pick an extra outbound
                 // peer from that network. The eviction logic in net_processing
                 // ensures that a peer from another network will be evicted.
-                std::tie(addr, addr_last_try) = addrman.Select(false, preferred_net);
+                if (preferred_net.has_value()) {
+                    networks = {*preferred_net};
+                }
+                std::tie(addr, addr_last_try) = addrman.Select(false, networks);
```

Comment:
This could be shorter (untested):

```cpp
std::tie(addr, addr_last_try) = preferred_net.has_value()
    ? addrman.Select(false, {*preferred_net})
    : addrman.Select(false, g_reachable_nets.All());
```